packages/core/src/baseCogniteClient.ts

```diff
@@ -201,7 +202,7 @@ export default class BaseCogniteClient {
   ) {
     const httpClient = new CDFHttpClient(
       getBaseUrl(baseUrl),
-      this.getRetryValidator()
+      options.retryValidator || this.getRetryValidator()
```

Comment:
I think adding this logic here might be confusing in the future. `getRetryValidator` would return a different fn while we use `options.retryValidator` for `CDFHttpClient`.

Instead, either `getRetryValidator` fn should return `options.retryValidator` with a fallback to `createUniversalRetryValidator` - or we can add a private variable (ex: `retryValidator`) inside the class and initialize it in constructor with the same logic (and `getRetryValidator` would return `this.retryValidator`).

Reply:
I like the second option, let me make the change.

Reply:
@cemreyavuz can you review again?